front/src/applications/operationalStudies/components/MacroEditor/osrdToNge.ts

```diff
@@ -30,14 +37,56 @@ import {
   type LabelDto,
 } from '../NGE/types';
 
+/**
+ * Get the TrainrunFrequencies from the TimetableItems.
+ * We need to add the unknown frequencies from the PacedTrains.
+ */
+const getNgeTrainrunFrequencies = (
+  state: MacroEditorState,
+  t: TFunction<'operationalStudies/scenario'>
+): TrainrunFrequency[] => {
+  // Get the default frequencies (TrainSchedule/30min/60min/120min)
+  const trainrunFrequencies = getDefaultTrainrunFrequencies(t);
+
+  // Add the unknown frequencies from the PacedTrains
+  state.timetableItems.forEach((timetableItem) => {
+    if (isPacedTrainResponseWithPacedTrainId(timetableItem)) {
+      const stepInMinutes = Duration.parse(timetableItem.paced.step).total('minute');
+      if (!trainrunFrequencies.find((f) => f.frequency === stepInMinutes)) {
+        const newFrequency: TrainrunFrequency = {
+          id: trainrunFrequencies.length + 1,
+          order: 0, // temporary order
+          frequency: stepInMinutes,
+          offset: 0,
+          name: t('macroEditor.stepXmin', { minutes: stepInMinutes }),
+          shortName: `${stepInMinutes}`,
+          linePatternRef: '60',
+        };
+        trainrunFrequencies.push(newFrequency);
+      }
+    }
+  });
+
+  // Sort and re-order the frequencies
+  trainrunFrequencies.sort((a, b) => a.frequency - b.frequency);
+  trainrunFrequencies.forEach((frequency, index) => {
+    frequency.order = index + 1;
+  });
+
+  return trainrunFrequencies;
+};
+
 /**
  * Execute the search payload and collect all result pages.
  */
 const executeSearch = async (
   state: MacroEditorState,
   dispatch: AppDispatch
 ): Promise<SearchResultItemOperationalPoint[]> => {
-  const pathSteps = state.trainSchedules.flatMap((ts) => ts.path);
+  const pathSteps: (PathItemLocation & {
+    deleted?: boolean;
+    id: string;
+  })[] = state.timetableItems.flatMap((timetableItem) => timetableItem.path);
```

Comment:
nit: maybe it's more scalable like that
```suggestion
  const pathSteps: TrainScheduleBase['path'] = state.timetableItems.flatMap((timetableItem) => timetableItem.path);
```

Reply:
It is !

Done in "front: use specific type for path"